### PR TITLE
Refactor accounts db repository - Closes #1442

### LINKED ISSUE #1442: Refactor accounts db repository
Parent: #1375

This is a quick follow-up on [PR-1312](https://github.com/LiskHQ/lisk/pull/1312) that was just merged, primarily to address what's apparent:

* Use of external SQL
* Use of proper database methods
* etc

### PR DESCRIPTION
**Pull request by vitaly-t**
* Closes #1442
* Closes the last cosmetic SQL issue for #1390

Also improves on some `blocks` repository methods, and how they are used within the module loader.

### CODE REVIEW

File: db/repos/accounts.js [hunk 1]
```diff
@@ -359,7 +351,7 @@ class AccountsRepo {
 				options.sortField.map((field, index) => {
 					sortSQL.push(this.pgp.as.format('$1:name $2:raw', [field, options.sortMethod[index]]));
 				});
-				sql = sql + `ORDER BY ${sortSQL.join(', ')}`;
```
Comment (nazarhussain) [inline db/repos/accounts.js]: @vitaly-t The case handles the situation where you sort on multiple fields. 
```
Select * from mem_accounts order by address ASC, isDelegate DESC;
```
Removing that `, ` will break the query.

Comment (vitaly-t) [reply db/repos/accounts.js]: I only removed `, `, because that's the default `.join`, and you do not need an extra space there ;)

**Comment by nazarhussain** [comment]
The changes looks good to scope the variables to module level. Apart from that I have some reservation on few overall patterns you are using. 

1. We had been using (or trying to use) a standard pattern to **double quote** every database field. No matter what its name. Now you are using a mixed pattern. If field name is lowercase you don't quote it. If field name contains any upper case letter, then you quote. it. I believe we should stick to one way and always enforce it by quoting every database field in query. 

2. You are converting every SQL function to lowercase e.g. `upper`, `encode`, while we were using every SQL related function/keyword with the upper case. I still believe uppercase convention is more suitable and increase readability of SQL statements.

Notice these two statements. 

```
SELECT array_agg("dependentId") FROM mem_accounts2u_delegates WHERE "accountId" = mem_accounts.address
````

```
SELECT ARRAY_AGG("dependentId") FROM "mem_accounts2u_delegates" WHERE "accountId" = "mem_accounts"."address"
````
Notice how mixed is the first example, some fields are quoted some are not. Some SQL related stuff is uppercase some in lowercase. While the second example have consistency around. 

@karmacoma As we are enforcing different standards/patterns/approaches as part of lisk-template, I believe we should decide on SQL writing approach as well.

**Comment by vitaly-t** [comment]
I am following a certain convention where one only uses upper-case for reserved words and constants, while function names are all lower-case.

Same for the column names, wrapping in double quotes only where needed, i.e. where column-case is used.

I've made it this way across all SQL in all repositories now. If we decide to go with a different standard, fine by me. But at least for now it is consistent :wink:

File: db/repos/accounts.js (hunk 1 above)
Comment (nazarhussain) [reply db/repos/accounts.js]: If its default for `join` then its ok. I was not aware that `join` have any default.

**Comment by karmacoma** [comment]
@nazarhussain You are correct, these are the patterns we had established thus far. Also agree, good idea to establish some guidelines so that whatever we decide on can be maintained properly.

@vitaly-t If it is indeed consistent now, then that should be good enough.

File: db/repos/accounts.js [hunk 2]
```diff
@@ -13,15 +13,57 @@
...
+// Only used in SELECT queries
+const dynamicFields = [
+	{name: 'rank', init: () => sql.columnRank},
+	{name: 'delegates', init: () => sql.columnDelegates},
+	{name: 'u_delegates', init: () => sql.columnUDelegates},
+	{name: 'multisignatures', init: () => sql.columnMultisignatures},
+	{name: 'u_multisignatures', init: () => sql.columnUMultisignatures}
+];
+
+const allFields = _.union(normalFields, immutableFields, dynamicFields);
```
Comment (nazarhussain) [inline db/repos/accounts.js]: 👍

**Pull request merged**